flybywiresim/a32nx
# feat(bleed): Fixed anti-ice fault logic

## Summary of Changes
Fixed logic for the ANTI-ICE FAULT light

## Screenshots (if necessary)


None

## References

None




## Additional context

Following the integration of the new WAI model, it has been noted that the FAULT light remains stuck when the wing anti-ice valve doesn't open and the WAI is eventually switched OFF. This was due to an if-statement that didn't cater for all the combinations.

Further to the above, the timer of the wing anti-ice test runs even if the system is unpressurised. This is against any logic, but this is how I interpret the schematics (30-11-01).


Discord username (if different from GitHub): pythonist#6593

## Testing instructions

Added unit test to cater for this case.

If you wish to verify, you can:
- Set the plane Cold and Dark
- Turn BATT ON
- Turn APU ON (No BLEED)
- Click WAI
- [ ]  Verify ON and FAULT light
- The FAULT light will not go away because pneumatics aren't pressurised.
- Wait for 30 sec
- Click WAI
- [ ]  Verify ON and FAULT lights disappear


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
